testing/ef-tests/src/cases/blockchain_test.rs

```diff
@@ -90,15 +92,23 @@ impl Case for BlockchainTestCase {
                     .database_provider_rw()
                     .unwrap();
 
+                let chain_spec_genesis_hash = chain_spec.genesis_hash();
+
                 // Insert initial test state into the provider.
-                provider.insert_historical_block(
-                    SealedBlock::<reth_primitives::Block>::from_sealed_parts(
-                        case.genesis_block_header.clone().into(),
-                        BlockBody::default(),
-                    )
-                    .try_recover()
-                    .unwrap(),
-                )?;
+                let genesis_block = SealedBlock::<reth_primitives::Block>::from_sealed_parts(
+                    case.genesis_block_header.clone().into(),
+                    BlockBody::default(),
+                )
+                .try_recover()
+                .unwrap();
+
+                assert_eq!(
+                    chain_spec_genesis_hash,
+                    genesis_block.hash_slow(),
+                    "shouldn't these hashes be the same?"
+                );
```

Comment:
I would have expected these to be the same, I get:

```
  left: 0xc0f91ac7900ffb638fdf6d1f20dc4a210294debe5bb4e9096390d852ab378304
 right: 0x8cbc69e33bd85b1f8d7bc6cae8f1d4502b74cfd0cd5f24a558c0d0c257c69daa
 ```

Reply:
Ah the chain_spec is derived from the ForkId, so it is using the default genesis block for that fork instead of the genesis block header in the json file

Reply:
From what I can see then:

- The execution spec tests define a genesis block header, and a network parameter (eg Cancun)
- There was code to convert the network parameter into a ChainSpec which has a genesis block based off of mainnet
- history_by_block_hash seemed to have been looking for the genesis block hash based off of the ChainSpec, whereas we saved the genesis block block hash that corresponds to the one in the execution spec test json file.
- I haven't checked, but I'm guessing latest() was looking at what we saved in the database and ignoring the ChainSpec